cortex/connection.py

```diff
@@ -25,47 +25,55 @@
 
 
 class ConnectionClient(_Client):
+    """A client used to manage connections on a Fabric server.
+
+    :param _Client: _description_
+    :type _Client: _type_
+    :return: An instance of a ConnectionClient
     """
-    A client used to manage connections.
-    """
-    URIs = {'connections': 'projects/{projectId}/connections'}
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-    def save_connection(self, connection: object):
-        """
-        Posts the connection client information.
-        :param connection: Connection object
-        :return: status
-        """
-        uri = self.URIs['connections'].format(projectId=self._project())
+
+    URIs = {"connections": "projects/{projectId}/connections"}
+
+    def save_connection(self, connection: dict) -> dict:
+        """Saves the provided `connection` dictionary to the Fabric API server.
+
+        :param connection: A Python dictionary containing specification for one of the `supported Fabric connection types <https://cognitivescale.github.io/cortex-fabric/docs/reference-guides/connection-types>`_
+        :type connection: dict
+        :return: A dictionary containing metadata about the saved connection
+        :rtype: dict
+        """  # pylint: disable=line-too-long
+        uri = self.URIs["connections"].format(projectId=self._project())
         data = json.dumps(connection)
-        headers = {'Content-Type': 'application/json'}
-        r = self._serviceconnector.request('POST', uri, data, headers)
-        raise_for_status_with_detail(r)
-        return r.json()
-
-    def get_connection(self, name: str):
-        """
-        Fetches a Connection to work with.
-        :param name: The name of the connection to retrieve.
-        :return: A Connection object.
-        """
-        port = os.getenv('CORTEX_CONNECTIONS_SERVICE_PORT_HTTP_CORTEX_CONNECTIONS') or '4450'
```

Comment:
Need a ticket for developer + secrets ..  How do developers access secrets on their laptop ? can we mark certain secrets a "exposed". or do we need an alternative like .env files

Reply:
Should we just start returning decrypted secrets like @mattsanchez has asked for in the past?